### DISCUSSION TITLE
怎么让机器人在掉线后自动登录回来？

### DISCUSSION THREAD

**Discussion by Silvigarabis** [iTXTech/mirai-console-loader#171] [Q&A] [answered]
我是在手机上的termux里运行的mcl，网络不是很稳定。
之前的版本里，如果网络不稳定或者直接没网，在恢复正常后，是会自己把机器人重新登录的
但是现在的版本只会这么提示
```
2023-02-28 14:21:36 V/Bot.**********: Event: BotOfflineEvent.Active(bot
=Bot(**********), cause=java.lang.IllegalStateException: NetworkHandler
 is already closed., reconnect=false)
2023-02-28 14:21:36 I/Bot.**********: Bot is closed manually with excep
tion: java.lang.IllegalStateException: NetworkHandler is already close
d.
java.lang.IllegalStateException: NetworkHandler is already closed.
```
我不清楚是什么原因导致现在不会在网络恢复后重新登录机器人，是我哪里弄错了吗？还是mcl出现了什么错误导致自动恢复登录被破坏。

**Comment by cssxsh** [comment, marked as answer]
https://github.com/mamoe/mirai/pull/2505

提供功能的是 mirai
mcl只是启动器

**Reply by Silvigarabis** [reply]
额，所以这个是mirai弄坏了自动恢复登录的功能吗？

**Reply by cssxsh** [reply]
弄坏的说法不准确，而且也没有所谓的 `自动恢复登录` , 准确的说是 `重新建立和服务器的连接` reconnect
以前是不管三七二十一，没连上服务器就死命连，同时这样也会导致一些问题
后来更新细致处理了一下，但是也导致网络不稳定的情况下被当成无网络可用

日志的表现为 `reconnect=false`

**Reply by Silvigarabis** [reply]
明白了，感谢解答
